workshops/templates/workshops/person_bulk_add_results.html

```diff
@@ -43,8 +43,19 @@
             </td>
             <td class="editable">
                 <span>{{ entry.role|default:"—" }}</span>
-                <input type="text" name="role" value="{{ entry.role|default:"" }}" class="hidden">
-                </td>
+                <select name="role" class="hidden">
+                {% if entry.role in possible_roles %}
+                    {% for role in possible_roles %}
+                    <option value="{{ role }}" {% if role == entry.role %}selected="selected"{% endif %}>{{ role }}</option>
+                    {% endfor %}
+                {% else %}
+                    <option selected="selected" value="{{ entry.role }}">{{ entry.role }}</option>
```

Comment:
Here's the logic: if user submits a valid option, I choose it. If they submit an invalid option, I show it them and tell them it's a wrong option.
![zrzut ekranu z 2016-11-28 18-47-27](https://cloud.githubusercontent.com/assets/72821/20679427/260953ee-b59b-11e6-9ea2-591f8047f1ef.png)